src/main/java/graphql/normalized/VariablePredicate.java

```diff
@@ -17,5 +18,25 @@ public interface VariablePredicate {
      *
      * @return true if a variable should be made
      */
-    boolean shouldMakeVariable(ExecutableNormalizedField executableNormalizedField, String argName, NormalizedInputValue normalizedInputValue);
+    boolean shouldMakeVariable(ExecutableNormalizedField executableNormalizedField,
+                               String argName,
+                               NormalizedInputValue normalizedInputValue);
+
+    /**
+     * Return true if a variable should be made for this query directive argument
+     * on the specified field
+     *
+     * @param executableNormalizedField the field in question
+     * @param queryAppliedDirective     the query directive in question
+     * @param argName                   the argument on the directive
+     * @param normalizedInputValue      the input value for that argument
+     *
+     * @return true if a variable should be made
+     */
+    default boolean shouldMakeVariable(ExecutableNormalizedField executableNormalizedField,
+                                       QueryAppliedDirective queryAppliedDirective,
+                                       String argName,
+                                       NormalizedInputValue normalizedInputValue) {
+        return false;
+    }
```

Comment:
since this is a SPI - we cant require them to implement this.  Or we could make it a breaking change say